thriftbp/client_pool.go

```diff
@@ -451,6 +452,22 @@ func newClientPool(
 			cfg.PoolGaugeInterval,
 			tags,
 		)
+
+		if err := prometheus.Register(clientPoolGaugeExporter{
```

Comment:
While I like the use of `Register()` over `MustRegister()`, this isn't going to be very effective in catching errors since `func (e clientPoolGaugeExporter) Describe()` is a noop.

I guess this is fine, since we can improve upon things later. But it's going to provide a misleading amount of safety as-is.

Reply:
I disagree. If the only choices we have is between this and panic (via `MustRegister`), I don't see how this is worse than panic.

Reply:
It's not worse, it's just not doing what you think it's doing.

Reply:
I can add a comment to say "this should not happen because desc is no-op but just in case" if that makes you happy.